modules/administration_manual/pages/configuration/files/encryption/enabling-user-key-encryption.adoc

```diff
@@ -0,0 +1,131 @@
+....
+sudo -u www-data php occ encryption:enable
+sudo -u www-data php occ encryption:select-encryption-type user-keys
+sudo -u www-data php occ encryption:encrypt-all
+sudo -u www-data php occ maintenance:singleuser --off
+....
+
+[[how-to-enable-users-file-recovery-keys]]
+== How To Enable User File Recovery Keys
+
+Once a user has encrypted their files, if they lose their ownCloud password, then they lose access to their encrypted files, as their files will be unrecoverable. 
+It is not possible, when user files are encrypted, to reset a user’s password using the standard reset process.
+If so, you’ll see a yellow banner warning:
+
+""
+Please provide an admin recovery password; otherwise, all user data will be lost.
+""
+
+To avoid all this, create a recovery key. 
+To do so, go to menu:Settings[Admin > encryption] and set a recovery key password.
+
+image:configuration/files/encryption10.png[image]
+
+You then need to ask your users to opt-in to the Recovery Key. 
+For the users to do this, they need to go to the ''**Personal**'' page and enable the recovery key. 
+This signals that they accept that the admin might have a way to decrypt their data for recovery reasons. 
+If they do _not_ do this, then the recovery key won’t work for them.
+
+image:configuration/files/encryption7.png[image]
+
+For users who have enabled password recovery, give them a new password and recover access to their encrypted files, by supplying the Recovery Key on the Users page.
+
+image:configuration/files/encryption8.png[image]
+
+You may change your recovery key password.
+
+image:configuration/files/encryption12.png[image]
+
+[NOTE]
+====
+Sharing a recovery key with a user group is *not* supported.
+This is only supported with xref:recreating-an-existing-master-key[the master key].
+====
+
+[[changing-the-recovery-key-password]]
+== Changing The Recovery Key Password
+
+If you have misplaced your recovery key password and need to replace it, here’s what you need to do:
+
+1. Delete the recovery key from both `data/owncloud_private_keys` and
+   `data/public-keys`
+2. Edit your database table `oc_appconfig` and remove the rows with the
+   config keys `recoveryKeyId` and `recoveryAdminEnabled` for the appid
+   `files_encryption`
+3. Login as admin and activate the recovery key again with a new
+   password. This will generate a new key pair
+4. All users who used the original recovery key will need to disable it
+   and enable it again. This deletes the old recovery share keys from their
+   files and encrypts their files with the new recovery key
+
+NOTE: You can only change the recovery key password if you know the original. 
+  This is by design, as only admins who know the recovery key password should be able to change it. 
+  If not, admins could hijack the recovery key from each other
+
+[IMPORTANT]
+====
+Replacing the recovery key will mean that all users will lose the possibility to recover their files until they have applied the new recovery key.
```

Comment:
Is there a description (maybe a link) available describing how/what to do to apply the new recovery key?

Reply:
I don't follow. Does the document not describe applying the recovery key?

Reply:
your are right, forget that comment